### PR TITLE
rename keys::{Secret, Public}

### LINKED ISSUE #59: Rename `frost::keys::{Secret, Public}`to indicate signing participants' shares, not full keys.
These are the signing pariticipant's secret share and corresponding public key, as opposed to the "full signature" signing key and verifying key. We have another type called SecretShare that has more metadata, hence not being named such, but we probably should rename them for clarity. This is somewhat low priority while the frost-core API remains unstable (by SemVer standards).

### PR DESCRIPTION
**Pull request by conradoplg**
Based on #105 

Closes #59 

Not sure about the names, these were what made sense to me, feel free to suggest other names.

### CODE REVIEW

**Comment by conradoplg** [comment]
Conflicts solved

File: frost-core/src/frost/keys.rs [hunk 1]
```diff
@@ -13,11 +13,11 @@ use zeroize::{DefaultIsZeroes, Zeroize};
 
 use crate::{frost::Identifier, Ciphersuite, Error, Field, Group, Scalar, VerifyingKey};
 
-/// A secret scalar value representing a signer's secret key.
+/// A secret scalar value representing a signer's secret share.
 #[derive(Clone, Copy, PartialEq, Eq)]
-pub struct Secret<C: Ciphersuite>(pub(crate) Scalar<C>);
+pub struct SecretShareValue<C: Ciphersuite>(pub(crate) Scalar<C>);
```
[the remark this answers is not in the document]
Comment (conradoplg) [reply frost-core/src/frost/keys.rs]: Looks good! There are some documentation warnings but I think some of those were already there, we can fix them all later.

**Comment by conradoplg** [comment]
@dconnolly your changes look good, is this ready to go?

**Review by dconnolly — APPROVED** [review]

**Comment by dconnolly** [comment]
> @dconnolly your changes look good, is this ready to go?

Yeah approved; I kept getting in analysis paralysis of trying to name things better, but also correspond accurately to the accepted language around shamir secret sharing, verifiable secret sharing (commitments), verifying shares/keys, etc etc, so might as well just merged for now :P

**Comment by conradoplg** [comment]
> > @dconnolly your changes look good, is this ready to go?
> 
> Yeah approved; I kept getting in analysis paralysis of trying to name things better, but also correspond accurately to the accepted language around shamir secret sharing, verifiable secret sharing (commitments), verifying shares/keys, etc etc, so might as well just merged for now :P

I can relate, we can have another pass at renaming stuff after we are up to date with the spec :grin:

**Pull request merged**